src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java

```diff
@@ -451,6 +481,200 @@ protected final void verify(Checker checker,
+    }
+
+    /**
+     * Tests the file with the check config.
+     *
+     * @param config check configuration.
+     * @param file input file path.
+     * @return list of actual violations.
+     * @throws Exception if exception occurs during verification process.
+     */
+    private List<String> getActualViolationsForFile(DefaultConfiguration config,
+          String file) throws Exception {
+        stream.flush();
+        stream.reset();
+        final List<File> files = Collections.singletonList(new File(file));
+        final Checker checker = createChecker(config);
+        final Map<String, List<String>> actualViolations =
+                getActualViolations(checker.process(files));
+        checker.destroy();
+        return actualViolations.getOrDefault(file, new ArrayList<>());
+    }
+
+    /**
+     * Returns the actual violations for each file that has been
+     * checked against google config modules.
+     * Each file is mapped to their corresponding violation messages. Reads input stream for these
+     * messages using instance of {@link InputStreamReader}.
+     *
+     * @param errorCount count of errors after checking set of files against {@link Checker}.
+     * @return a {@link Map} object containing file names and the corresponding violation messages.
+     * @throws IOException exception can occur when reading input stream.
+     */
+    private Map<String, List<String>> getActualViolations(int errorCount) throws IOException {
+        // process each of the lines
+        try (ByteArrayInputStream inputStream =
+                     new ByteArrayInputStream(stream.toByteArray());
+             LineNumberReader lnr = new LineNumberReader(
+                     new InputStreamReader(inputStream, StandardCharsets.UTF_8))) {
+            final Map<String, List<String>> actualViolations = new HashMap<>();
+            for (String line = lnr.readLine(); line != null && lnr.getLineNumber() <= errorCount;
+                 line = lnr.readLine()) {
+                // have at least 2 characters before the splitting colon,
+                // to not split after the drive letter on Windows
+                final String[] actualViolation = line.split("(?<=.{2}):", 2);
+                final String actualViolationFileName = actualViolation[0];
+                final String actualViolationMessage = actualViolation[1];
+
+                actualViolations
+                        .computeIfAbsent(actualViolationFileName, key -> new ArrayList<>())
+                        .add(actualViolationMessage);
+            }
+
+            return actualViolations;
+        }
+    }
+
+    /**
+     * Creates a google configuration based on the list of modules.
+     *
+     * @param listOfModules list of modules to load from google config.
+     * @return google configuration based on the list of modules.
+     */
+    private static DefaultConfiguration createGoogleConfig(Map<String, String[]> listOfModules) {
```

Comment:
createConfigForModulesOfGoogleConfig

Reply:
done.

https://github.com/checkstyle/checkstyle/blob/a0f82691156c959fbb7dccd991c3e5c4f83022e1/src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java#L560-L564